lib/irb/source_finder.rb

```diff
@@ -26,6 +31,13 @@ def find_source(signature)
       when /\A(?<owner>[A-Z]\w*(::[A-Z]\w*)*)#(?<method>[^ :.]+)\z/ # Class#method
         owner = eval(Regexp.last_match[:owner], context_binding)
         method = Regexp.last_match[:method]
+        if esses
+          super_owner = owner
+          s_count.times {|s| super_owner = super_owner.superclass}
```

Comment:
if there are too many -ssss, `super_owner.superclass` raises ``undefined method `superclass' for nil:NilClass``

Reply:
I added this test in there already in the previous commit:
```ruby
    def test_show_source_method_exceeds_super_chain
      code = <<~RUBY
        class Baz
          def foo
          end
        end
      
        class Bar < Baz
          def foo
            super
          end
        end
      RUBY
      File.write("#{@tmpdir}/bazbar.rb", code)
      out, err = execute_lines(
        "irb_load '#{@tmpdir}/bazbar.rb'\n",
        "show_source Bar#foo -ss",
      )
      assert_match(%r[Error: Couldn't locate a definition for Bar#foo -ss\n=> nil\n], out)
    end
```
Does that cover the use case you're referring to? If not I'm happy to add another.